src/SampleUtterances.ts

```diff
@@ -93,8 +93,22 @@ export class SamplePhrase {
         const startIndex = phrase.indexOf("{");
         if (startIndex !== -1) {
             const endIndex = phrase.indexOf("}", startIndex);
-            this.slotNames.push(phrase.substring(startIndex + 1, endIndex));
-            phrase = phrase.substring(0, startIndex).trim() + "(.*)" + phrase.substring(endIndex + 1).trim();
+            const slotName = phrase.substring(startIndex + 1, endIndex);
+
+            const pipe = slotName.indexOf("|");
+
+            if(pipe === -1){
+                this.slotNames.push(slotName);
+                phrase = phrase.substring(0, startIndex).trim() + "(.*)" + phrase.substring(endIndex + 1).trim();
+            } else{
+                // Literal are in the format "sample { <literal sample> | <slotname>}"
+                // e.g.: "I'm an {aquarius | literal}"
+                const literalSample = slotName.substring(0, pipe);
+                const literalSlotName = slotName.substring(pipe + 2, slotName.length);
+                this.slotNames.push(literalSlotName);
+                phrase = phrase.substring(0, startIndex).trim() + literalSample.trim() + phrase.substring(endIndex + 1).trim();
```

Comment:
This is going to make the phrase match ONLY if the example literal value is in the utterance - I believe it is better not to do this (i.e., it will only match if the utterance is "second literal input"). On the one hand, this will improve the performance when the suggested phrases are used. But our logic is that we will then restrict it to ONLY those enumerated phrases, so no other phrases will work.

Our matching algorithm is based on looking at how precisely the text of the phrase as well as the values of the slots are matched. It is simplistic compared to a real NLU. But it should perform better, and for a greater variety of utterances, if this is left out.

So long story short, this line should be taken out, and the phrase on line 102 should be moved outside the if condition (as it will be parsed the same way for both).